File: server/static/sdks/nodejs/joystauth.js
const https = require('https');
const http = require('http');
const os = require('os');
const { execSync } = require('child_process');

class JoystAuth {
    constructor(name, token, version = "1.0", url = "https://joystauth.cc") {
        this.name = name;
        this.token = token;
        this.version = version;
        this.url = (url || "https://joystauth.cc").replace(/\/+$/, '');
        this.sessionid = null;
        this.hwid = this.getHWID();
        this.userData = { username: "", subscription: "default", expiry: "Lifetime", hwid: this.hwid };
        this.response = { success: false, message: "" };

        // ⚡ 1. Inbuilt Auto-Init
        this.init(true);

        // ⚡ 2. Inbuilt Live Heartbeat Watchdog
        this.startHeartbeatWatchdog();
    }

    getHWID() {
        try {
            if (process.platform === 'win32') {
                const out = execSync('whoami /user', { stdio: ['pipe', 'pipe', 'ignore'] }).toString();
                const match = out.match(/S-1-5-21-\d+-\d+-\d+-\d+/);
                if (match) return match[0];
            }
        } catch(e) {}
        return os.hostname() + "_" + os.userInfo().username;
    }

    _post(endpoint, data) {
        return new Promise((resolve) => {
            const urlObj = new URL(this.url + endpoint);
            const isHttps = urlObj.protocol === 'https:';
            const client = isHttps ? https : http;
            const body = JSON.stringify(data);

            const req = client.request(urlObj, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'Content-Length': Buffer.byteLength(body),
                    'User-Agent': 'JoystEnclave-Node/2.0'
                }
            }, (res) => {
                let chunks = '';
                res.on('data', c => chunks += c);
                res.on('end', () => {
                    try { resolve(JSON.parse(chunks)); }
                    catch(e) { resolve({ success: false, message: "Invalid JSON response" }); }
                });
            });

            req.on('error', (e) => resolve({ success: false, message: e.message }));
            req.write(body);
            req.end();
        });
    }

    async init(autoExitOnMaint = true) {
        const res = await this._post('/api/v1/client/init', {
            app_name: this.name,
            app_token: this.token,
            version: this.version,
            hwid: this.hwid
        });

        if (res.success) {
            this.sessionid = res.sessionid;
            this.response = { success: true, message: res.message || "Initialized" };
            return true;
        } else {
            const msg = res.message || res.detail || "Authentication server error.";
            this.response = { success: false, message: msg };
            if (autoExitOnMaint) {
                console.error(`\n🚨 [JOYST ALERT] ${msg}\n`);
                process.exit(1);
            }
            return false;
        }
    }

    startHeartbeatWatchdog() {
        setInterval(async () => {
            if (!this.sessionid) return;
            try {
                const res = await this._post('/api/v1/client/check', {
                    app_name: this.name,
                    app_token: this.token,
                    hwid: this.hwid,
                    username: this.userData.username,
                    sessionid: this.sessionid
                });
                if (res && res.success === false) {
                    console.error(`\n🚨 [JOYST SECURITY ALERT] ${res.message || "Application placed into maintenance."}\n`);
                    process.exit(1);
                }
            } catch(e) {}
        }, 15000);
    }

    async login(username, password) {
        const res = await this._post('/api/v1/client/login', {
            app_name: this.name,
            app_token: this.token,
            username: username.trim(),
            password: password,
            hwid: this.hwid,
            sessionid: this.sessionid
        });

        if (res.success) {
            this.userData.username = username;
            this.userData.subscription = res.subscription || "default";
            this.userData.expiry = res.expires_at || "Lifetime";
            this.response = { success: true, message: res.message || "Logged in" };
            return true;
        } else {
            this.response = { success: false, message: res.message || "Login failed" };
            return false;
        }
    }

    async license(key) {
        const res = await this._post('/api/v1/client/license', {
            app_name: this.name,
            app_token: this.token,
            license_key: key.trim(),
            hwid: this.hwid,
            sessionid: this.sessionid
        });

        if (res.success) {
            this.userData.username = res.username || key;
            this.userData.subscription = res.subscription || "VIP Tier";
            this.userData.expiry = res.expires_at || "Lifetime";
            this.response = { success: true, message: res.message || "License Active" };
            return true;
        } else {
            this.response = { success: false, message: res.message || "Invalid License" };
            return false;
        }
    }

    async register(username, password, licenseKey) {
        const res = await this._post('/api/v1/client/register', {
            app_name: this.name,
            app_token: this.token,
            username: username.trim(),
            password: password,
            license_key: licenseKey.trim(),
            hwid: this.hwid,
            sessionid: this.sessionid
        });

        if (res.success) {
            this.userData.username = username;
            this.response = { success: true, message: res.message || "Registered" };
            return true;
        } else {
            this.response = { success: false, message: res.message || "Registration failed" };
            return false;
        }
    }

    async var(varName) {
        const res = await this._post('/api/v1/client/var', {
            app_name: this.name,
            app_token: this.token,
            var_name: varName,
            sessionid: this.sessionid
        });
        return res.value || "";
    }
}

module.exports = JoystAuth;

File: server/static/sdks/go/joystauth.go
package joystauth

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
	"os/exec"
	"regexp"
	"strings"
	"time"
)

type UserData struct {
	Username     string `json:"username"`
	Subscription string `json:"subscription"`
	Expiry       string `json:"expiry"`
	HWID         string `json:"hwid"`
}

type ResponseData struct {
	Success bool   `json:"success"`
	Message string `json:"message"`
}

type API struct {
	Name      string
	Token     string
	Version   string
	URL       string
	HWID      string
	SessionID string
	UserData  UserData
	Response  ResponseData
	client    *http.Client
}

func New(name, token string, version ...string) *API {
	ver := "1.0"
	if len(version) > 0 {
		ver = version[0]
	}
	api := &API{
		Name:    name,
		Token:   token,
		Version: ver,
		URL:     "https://joystauth.cc",
		HWID:    getHWID(),
		client:  &http.Client{Timeout: 8 * time.Second},
	}
	api.Init(true)
	return api
}

func getHWID() string {
	out, err := exec.Command("whoami", "/user").Output()
	if err == nil {
		re := regexp.MustCompile(`S-1-5-21-\d+-\d+-\d+-\d+`)
		match := re.FindString(string(out))
		if match != "" {
			return match
		}
	}
	hostname, _ := os.Hostname()
	return hostname
}

func (a *API) post(endpoint string, data map[string]interface{}) (map[string]interface{}, error) {
	jsonData, _ := json.Marshal(data)
	resp, err := a.client.Post(a.URL+endpoint, "application/json", bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var res map[string]interface{}
	json.NewDecoder(resp.Body).Decode(&res)
	return res, nil
}

func (a *API) Init(autoExitOnMaint bool) bool {
	res, err := a.post("/api/v1/client/init", map[string]interface{}{
		"app_name":  a.Name,
		"app_token": a.Token,
		"version":   a.Version,
		"hwid":      a.HWID,
	})
	if err != nil || res == nil {
		a.Response = ResponseData{Success: false, Message: "Server connection failed"}
		return false
	}

	if ok, _ := res["success"].(bool); ok {
		if sid, exists := res["sessionid"].(string); exists {
			a.SessionID = sid
		}
		a.Response = ResponseData{Success: true, Message: "Initialized"}
		return true
	}

	msg, _ := res["message"].(string)
	a.Response = ResponseData{Success: false, Message: msg}
	if autoExitOnMaint && strings.Contains(strings.ToLower(msg), "maintenance") {
		fmt.Println("[JOYST ALERT] " + msg)
		os.Exit(1)
	}
	return false
}

func (a *API) Login(username, password string) bool {
	res, err := a.post("/api/v1/client/login", map[string]interface{}{
		"app_name":  a.Name,
		"app_token": a.Token,
		"username":  strings.TrimSpace(username),
		"password":  password,
		"hwid":      a.HWID,
		"sessionid": a.SessionID,
	})
	if err != nil || res == nil {
		a.Response = ResponseData{Success: false, Message: "Connection error"}
		return false
	}

	if ok, _ := res["success"].(bool); ok {
		a.UserData.Username = username
		a.Response = ResponseData{Success: true, Message: "Logged in successfully"}
		return true
	}

	msg, _ := res["message"].(string)
	a.Response = ResponseData{Success: false, Message: msg}
	return false
}

func (a *API) License(key string) bool {
	res, err := a.post("/api/v1/client/license", map[string]interface{}{
		"app_name":    a.Name,
		"app_token":   a.Token,
		"license_key": strings.TrimSpace(key),
		"hwid":        a.HWID,
		"sessionid":   a.SessionID,
	})
	if err != nil || res == nil {
		a.Response = ResponseData{Success: false, Message: "Connection error"}
		return false
	}

	if ok, _ := res["success"].(bool); ok {
		a.UserData.Username = key
		a.Response = ResponseData{Success: true, Message: "License active"}
		return true
	}

	msg, _ := res["message"].(string)
	a.Response = ResponseData{Success: false, Message: msg}
	return false
}
